src/tribol/search/InterfacePairFinder.cpp

```diff
@@ -1,643 +1,873 @@
-   ElemVertRelation m_elemVertConnectivity;
-};
+///////////////////////////////////////////////////////////////////////////////
 
 /*!
- * \brief Helper class to compute the candidate pairs for a coupling scheme
+ * \brief Implicit Grid helper class to compute the candidate pairs for a coupling scheme
  *
  * A GridSearch indexes the elements from the first mesh of
  * the coupling scheme in a spatial index that requires element bounding boxes.
  * Then, for each of the elements in the second mesh, we find the candidate
  * pairs and add them to the coupling scheme's list of contact pairs.
  *
- * The spatial index is generated in \a generateSpatialIndex()
+ * The spatial index is generated in \a initialize()
  * and the search is performed in \a findInterfacePairs()
  *
  * \tparam D The spatial dimension of the coupling scheme mesh vertices.
  */
 template<int D>
-class GridSearch
+class GridSearch : public SearchBase
 {
 public:
-   using ImplicitGridType = spin::ImplicitGrid<D, axom::SEQ_EXEC, int>;
-   using SpacePoint = typename ImplicitGridType::SpacePoint;
-   using SpaceVec = typename ImplicitGridType::SpaceVec;
-   using SpatialBoundingBox = typename ImplicitGridType::SpatialBoundingBox;
-
-   /*!
-    * Constructs a GridSearch instance over CouplingScheme \a couplingScheme
-    * \pre couplingScheme is not null
-    */
-   GridSearch(CouplingScheme* couplingScheme)
-      : m_couplingScheme(couplingScheme)
-   {
-      MeshManager & meshManager = MeshManager::getInstance();
+  using BBox = primal::BoundingBox<RealT, D>;
+  using PointT = primal::Point<RealT, D>;
+
+  using ImplicitGridType = spin::ImplicitGrid<D,axom::SEQ_EXEC,int>;
+  using SpacePoint = typename ImplicitGridType::SpacePoint;
+  using SpaceVec = typename ImplicitGridType::SpaceVec;
+  using SpatialBoundingBox = typename ImplicitGridType::SpatialBoundingBox;
+
+  /*!
+   * Constructs a GridSearch instance over CouplingScheme \a couplingScheme
+   * \pre couplingScheme is not null
+   */
+  GridSearch( CouplingScheme* couplingScheme )
+    : m_coupling_scheme( couplingScheme )
+    , m_mesh1( m_coupling_scheme->getMesh1() )
+    , m_mesh2( m_coupling_scheme->getMesh2() )
+  {}
+
+  /*!
+   * Constructs spatial index over elements of coupling scheme's first mesh
+   */
+  void initialize() override
+  {
+    // TODO does this tolerance need to scale with the mesh?
+    const RealT bboxTolerance = 1e-6;
+
+    m_coupling_scheme->getInterfacePairs().clear();
+
+    // Find the bounding boxes of the elements in the first mesh
+    // Store them in an array for efficient reuse
+    m_gridBBox.clear();
+    m_meshBBoxes1.reserve(m_mesh1.numberOfElements());
+    for(int i=0; i< m_mesh1.numberOfElements(); ++i)
+    {
+      m_meshBBoxes1.emplace_back(elementBoundingBox(m_mesh1, i));
+    }
 
-      integer meshId1 = m_couplingScheme->getMeshId1();
-      MeshData const & meshData1 = meshManager.GetMeshInstance(meshId1);
-      m_meshWrapper1 = MeshWrapper<D>(&meshData1);
```

Comment:
Looks like the meshWrapper was used in the implicit grid `initialize()` function. If I'm seeing it correctly you moved some of the meshWrapper stuff onto the GridSearch class itself?